docs(list): update list documentation with the new "selected" property (#6081)
IgniteUI/igniteui-docfx: en/components/list.md

## List Item Selection

> The list items have a `selected` state property that helps us track which items have been selected. This property allows us to identify and manage the selection status of each item.

Reviewer: ```suggestion
The list items have a `selected` property that helps us track which items are "selected". This property allows us to identify and manage the selection status of each item.
```

> Here's an example, in which by using the `selected` property we apply a background color to the list items that we select:

Reviewer: ```suggestion
Here's an example illustrating how the visual style of the items changes when using the `selected` property:
```

<code-view style="height: 420px"
           data-demos-base-url="{environment:demosBaseUrl}"
           iframe-src="{environment:demosBaseUrl}/lists/list-item-selection" >
</code-view>

> To make list items selectable, we simply need to add the `selected` property value to each list item we want to be able to select. By default, this property will be set to `false`. Then, we add a `(click)` event to each list item, which will toggle the `selected` property value between `true` and `false`, effectively switching the selection state each time the item is clicked."

Reviewer: ```suggestion
By default, the `selected` property is set to `false`. We can toggle its value using an inline expression bound to the `(click)` event on each list item, effectively switching the visual state of the item each time it's clicked."
```

```html
<igx-list>
    <igx-list-item [isHeader]="true">Contacts</igx-list-item>
    @for (contact of contacts | igxFilter: filterContacts; track contact) {
      <igx-list-item [selected]="contact.selected" (click)="contact.selected = !contact.selected">
        <igx-avatar igxListThumbnail [src]="contact.photo" shape="circle"></igx-avatar>
        <span igxListLineTitle>{{ contact.name }}</span>
        <span igxListLineSubTitle>{{ contact.phone }}</span>
        <igx-icon igxListAction [style.color]="contact.isFavorite ? 'orange' : 'lightgray'" igxRipple="pink"
          [igxRippleCentered]="true" (click)="toggleFavorite(contact, $event)"
        (mousedown)="mousedown($event)">star</igx-icon>
      </igx-list-item>
    }
  </igx-list>
```

When a list item is selected, the class `.igx-list__item-base--selected` is added to the element. We can use this class to style the selected state of the element.

```scss
.igx-list__item-base--selected .igx-list__item-content {
  background-color: var(--ig-secondary-500);
}

> ```

Reviewer: We shouldn't be telling people to use the `.igx-list__item-base--selected` class to style items when they are marked as selected. Rather, there are new CSS variables that allow us to manage that.

Here's a list of all CSS variables that can be used to alter an item when it has the `selected` property added to it:

```css
/* 
--item-background-selected
--item-text-color-selected
--item-title-color-selected
--item-action-color-selected
--item-subtitle-color-selected
--item-thumbnail-color-selected
*/

/* 
You can give an example using at least these three custom properties (not with the colors I've suggested bellow):
*/

igx-list-item {
  --item-background-selected: orange;
  --item-text-color-selected: black;
  --item-title-color-selected: black;
}
```
